src/plugins/console/server/lib/proxy_request.test.ts

```diff
@@ -115,39 +116,33 @@ describe(`Console's send request`, () => {
       } as any;
     });
 
-    it('should decode percent-encoded uri pathname and encode it correctly', async () => {
-      const uri = new URL(
-        `http://noone.nowhere.none/%{[@metadata][beat]}-%{[@metadata][version]}-2020.08.23`
-      );
-      const result = await proxyRequest({
-        agent: null as any,
-        headers: {},
-        method: 'get',
-        payload: null as any,
-        timeout: 30000,
+    const runEncodePathTests = async (initialPath: string, expectedPath: string, uri?: URL) => {
+      const result = await sendProxyRequest({
+        requestPath: initialPath,
         uri,
       });
-
       expect(result).toEqual('done');
       const [httpRequestOptions] = stub.firstCall.args;
-      expect((httpRequestOptions as any).path).toEqual(
-        '/%25%7B%5B%40metadata%5D%5Bbeat%5D%7D-%25%7B%5B%40metadata%5D%5Bversion%5D%7D-2020.08.23'
+      expect((httpRequestOptions as any).path).toEqual(expectedPath);
+    };
+
+    it('should correctly encode invalid URL characters included in path', async () => {
+      await runEncodePathTests(
```

Comment:
What do you think about updating this util to accept an object? It might be easier to scan in the future which args represent the input vs. expected result. For example:

```
 await runEncodePathTests({
        initialPath: '_index/.test',
        uri: new URL('http://noone.nowhere.none/_index/.test?q=something&v=something'),
        expectedPath: '_index/.test?q=something&v=something',
      });
```